src/legacy/server/sample_data/usage/usage.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as Hapi from 'hapi';
+
+const SAVED_OBJECT_ID = 'sample-data-telemetry';
+
+export function usage(request: Hapi.Request) {
```

Comment:
Why not just export 2 distinct functions instead to make things clearer? Burying the used methods in the return of `usage` makes the flow harder to pick out.

```
function getInternalRepsitory(server) {
  const {
    savedObjects: { getSavedObjectsRepository },
  } = server;
  const { callWithInternalUser } = server.plugins.elasticsearch.getCluster('admin');
  return getSavedObjectsRepository(callWithInternalUser);
}

const handleIncrementError = (server, err: Error) => {
  if (err != null) {
    server.log(['debug', 'sample_data', 'telemetry'], err.stack);
  }
  server.log(
    ['warning', 'sample_data', 'telemetry'],
    `saved objects repository incrementCounter encountered an error: ${err}`
  );
};

export async function incrInstallUsage(request, sampleDataSetId) {
  try {
    const internalRepository = getInternalRepsitory(request.server);
    internalRepository.incrementCounter(SAVED_OBJECT_ID, dataSet, `installCount`);
  } catch (err) {
    handleIncrementError(request.server, err);
  }
}

export async function incrUninstallUsage(request, sampleDataSetId) {
  try {
    const internalRepository = getInternalRepsitory(request.server);
    internalRepository.incrementCounter(SAVED_OBJECT_ID, dataSet, `unInstallCount`);
  } catch (err) {
    handleIncrementError(request.server, err);
  }
}
```

Reply:
I'm a bit `-1` for this bc I'm not seeing the original code as burying the return: it's a typical Revealing Module pattern and pretty old school.

I also prefer the way I have it as far as the caller's syntax:
    usage(request).addInstall(params.id);
    usage(request).addUninstall(params.id);

How strongly do you feel about it?

Reply:
Its your PR so you get to make the final call. I felt strongly enough to bring it up.